src/pages/studyView/charts/categoryBarChart/CategoryBarChart.tsx

```diff
@@ -0,0 +1,353 @@
+                        }}
+                        height={this.props.height - this.bottomPadding}
+                        width={this.props.width}
+                        padding={{
+                            left: 40,
+                            right: 20,
+                            top: 10,
+                            bottom: this.bottomPadding,
+                        }}
+                        theme={VICTORY_THEME}
+                    >
+                        <VictoryAxis
+                            tickValues={this.tickValuesBasedOnNA}
+                            tickFormat={(t: number) => this.tickFormat[t - 1]}
+                            domain={[0, this.maximumX]}
+                            tickLabelComponent={<CategoryBarChartAxisLabel />}
+                            style={{
+                                tickLabels: {
+                                    angle: TILT_ANGLE,
+                                    verticalAnchor: 'start',
+                                    textAnchor: 'start',
+                                },
+                            }}
+                        />
+                        <VictoryAxis
+                            dependentAxis={true}
+                            tickFormat={(t: number) =>
+                                Number.isInteger(t) ? t.toFixed(0) : ''
+                            }
+                        />
+                        <VictoryBar
+                            barRatio={this.barRatio}
+                            style={{
+                                data: {
+                                    fill: (d: BarDatum) =>
+                                        isDataBinSelected(
+                                            d.dataBin,
+                                            this.props.filters
+                                        ) || this.props.filters.length === 0
+                                            ? STUDY_VIEW_CONFIG.colors.theme
+                                                  .primary
+                                            : DEFAULT_NA_COLOR,
+                                },
+                            }}
+                            data={this.barDataBasedOnNA}
+                            events={this.barPlotEvents}
+                        />
+                        {this.labelShowingNA}
+                    </VictoryChart>
+                )}
+                {ReactDOM.createPortal(
+                    <CategoryBarChartToolTip
+                        mousePosition={this.mousePosition}
+                        windowWidth={WindowStore.size.width}
+                        model={this.toolTipModel}
+                        totalBars={this.barData.length}
+                        currentBarIndex={this.currentBarIndex}
+                    />,
+                    document.body
+                )}
+            </div>
+        );
+    }
+}
+
+const onlyContainsNA = (element: BarDatum): boolean => {
```

Comment:
like said above, these functions are in BarChart as well. we could make export these functions to prevent duplication